egs/hkust/s5/local/hkust_data_prep.sh

```diff
@@ -1,97 +1,74 @@
 
-
 #Transcriptions preparation
 
 #collect all trans, convert encodings to utf-8,
-find $HKUST_TEXT_DIR -iname "*.txt" | grep -i "trans/train" | xargs cat |\
+find $hkust_text_dir -iname "*.txt" | grep -i "trans/train" | xargs cat |\
   iconv -f GBK -t utf-8 - | perl -e '
     while (<STDIN>) {
       @A = split(" ", $_);
       if (@A <= 1) { next; }
-      if ($A[0] eq "#") { $utt_id = $A[1]; } 
+      if ($A[0] eq "#") { $utt_id = $A[1]; }
       if (@A >= 3) {
-        $A[2] =~ s:^([AB])\:$:$1:; 
-        printf "%s-%s-%06.0f-%06.0f", $utt_id, $A[2], 100*$A[0] + 0.5, 100*$A[1] + 0.5; 
-        for($n = 3; $n < @A; $n++) { print " $A[$n]" }; 
-        print "\n"; 
+        $A[2] =~ s:^([AB])\:$:$1:;
+        printf "%s-%s-%06.0f-%06.0f", $utt_id, $A[2], 100*$A[0] + 0.5, 100*$A[1] + 0.5;
+        for($n = 3; $n < @A; $n++) { print " $A[$n]" };
+        print "\n";
       }
     }
-  ' | sort -k1 > $train_dir/transcripts.txt 
+  ' | sort -k1 > $train_dir/transcripts.txt || exit 1;
 
-find $HKUST_TEXT_DIR -iname "*.txt" | grep -i "trans/dev" | xargs cat |\
+find $hkust_text_dir -iname "*.txt" | grep -i "trans/dev" | xargs cat |\
   iconv -f GBK -t utf-8 - | perl -e '
     while (<STDIN>) {
       @A = split(" ", $_);
       if (@A <= 1) { next; }
-      if ($A[0] eq "#") { $utt_id = $A[1]; } 
+      if ($A[0] eq "#") { $utt_id = $A[1]; }
       if (@A >= 3) {
-        $A[2] =~ s:^([AB])\:$:$1:; 
-        printf "%s-%s-%06.0f-%06.0f", $utt_id, $A[2], 100*$A[0] + 0.5, 100*$A[1] + 0.5; 
-        for($n = 3; $n < @A; $n++) { print " $A[$n]" }; 
-        print "\n"; 
+        $A[2] =~ s:^([AB])\:$:$1:;
+        printf "%s-%s-%06.0f-%06.0f", $utt_id, $A[2], 100*$A[0] + 0.5, 100*$A[1] + 0.5;
+        for($n = 3; $n < @A; $n++) { print " $A[$n]" };
+        print "\n";
       }
     }
-  ' | sort -k1  > $dev_dir/transcripts.txt
+  ' | sort -k1  > $dev_dir/transcripts.txt || exit 1;
 
-
-
-#transcripts normalization and segmentation 
+#transcripts normalization and segmentation
 #(this needs external tools),
-#Download and configure segment tools   
-pyver=`python --version 2>&1 | sed -e 's:.*\([2-3]\.[0-9]\+\).*:\1:g'`
-export PYTHONPATH=$PYTHONPATH:`pwd`/tools/mmseg-1.3.0/lib/python${pyver}/site-packages
-if [ ! -d tools/mmseg-1.3.0/lib/python${pyver}/site-packages ]; then
-  echo "--- Downloading mmseg-1.3.0 ..."
-  echo "NOTE: it assumes that you have Python, Setuptools installed on your system!"
-  wget -P tools http://pypi.python.org/packages/source/m/mmseg/mmseg-1.3.0.tar.gz 
-  tar xf tools/mmseg-1.3.0.tar.gz -C tools
-  cd tools/mmseg-1.3.0
-  mkdir -p lib/python${pyver}/site-packages
-  python setup.py build 
-  python setup.py install --prefix=.
-  cd ../..
-  if [ ! -d tools/mmseg-1.3.0/lib/python${pyver}/site-packages ]; then
-    echo "mmseg is not found - installation failed?"
-    exit 1
-  fi
+if [ ! `echo $PYTHONPATH | grep mmseg` ]; then
+  echo "mmseg is not found. Checkout tools/extra/install_mmseg.sh"
+  exit 1
```

Comment:
Perhaps a less brittle check would be a simple oneliner that tries to import given module
something like (probably incorrect syntactically)
python -c "import sys; try: import mmseg; except: sys.exit(1);"